diff --git a/sonata/config.py b/sonata/config.py
index 8b53b4b..19b8532 100644
--- a/sonata/config.py
+++ b/sonata/config.py
@@ -86,6 +86,7 @@ class Config:
             'audioscrobbler': {
                 'as_enabled': ('use_audioscrobbler', 'boolean', False),
                 'as_password_md5': ('password_md5', '', ''),
+                'as_password_md5_file': ('password_md5_file', '', ''),
                 'as_username': ('username', '', '')},
             'connection': {
                 'autoconnect': ('auto', 'boolean', True),
diff --git a/sonata/scrobbler.py b/sonata/scrobbler.py
index 3f5da8b..941ecc9 100644
--- a/sonata/scrobbler.py
+++ b/sonata/scrobbler.py
@@ -67,9 +67,17 @@ class Scrobbler:
 
     def init(self):
         """Initialize the Audioscrobbler support if enabled and configured"""
-        if audioscrobbler is not None and self.config.as_enabled and \
-           len(self.config.as_username) > 0 and \
-           len(self.config.as_password_md5) > 0:
+        if audioscrobbler is None or not self.config.as_enabled:
+            return
+
+        if len(self.config.as_username) == 0:
+            return
+
+        if len(self.config.as_password_md5_file) > 0:
+            with open(self.config.as_password_md5_file, 'r') as f:
+                self.config.as_password_md5 = f.readline().rstrip('\n')
+
+        if len(self.config.as_password_md5) > 0:
             thread = threading.Thread(target=self.init_thread)
             thread.name = "Scrobbler"
             thread.daemon = True
